src/thunks.js

```diff
@@ -320,11 +322,51 @@ export function fetchAllInvoices() {
     }
     const url = '/api/invoices/';
     return Request
-            .get(url)
-            .set('Authorization', 'Bearer ' + authToken)
-            .then((response) => {
-              const normalizedInvoices = normalizeArray(response.body);
-              dispatch(receiveAllInvoices(normalizedInvoices));
-            })
+      .get(url)
+      .set('Authorization', 'Bearer ' + authToken)
+      .then((response) => {
+        const normalizedInvoices = normalizeArray(response.body);
+        dispatch(receiveInvoices(normalizedInvoices));
+      })
+  }
+}
+
+export function getInvoice(id) {
+  return (dispatch) => {
+    const authToken = getAuthToken();
+    if (!authToken) {
+      return;
+    }
+    const url = '/api/invoices/' + id + '/';
+    dispatch(requestInvoice(id));
+    return Request
+      .get(url)
+      .set('Authorization', 'Bearer ' + authToken)
+      .then((response) => {
+        const normalizedInvoices = normalizeArray([response.body]);
+        dispatch(receiveInvoices(normalizedInvoices));
+      })
   }
 }
+
+export function markInvoicePaid(id) {
+  return (dispatch) => {
+    const authToken = getAuthToken();
+    if (!authToken) {
+      return;
```

Comment:
Let's do away with this bad habit of failing silently in our thunks. In this case, it makes sense to `dispatch(invoicePaymentFailed());` and then `return`, rather than `return` silently.